NanoXLSX/Worksheet.cs

```diff
@@ -2629,5 +2629,123 @@ private static bool WorksheetExists(string name, Workbook workbook)
+                Address newAddress = new Address(col, row + numberOfNewRows);
+
+                Cell newCell = new Cell(cell.Value.Value, cell.Value.DataType, newAddress);
+                if (cell.Value.CellStyle != null)
+                {
+                    newCell.SetStyle(cell.Value.CellStyle); // Apply the style from the "old" cell.
+                } 
+                newCells.Add(newAddress.GetAddress(), newCell);
+
+                // Delete the original cells since the key cannot be changed.
+                this.Cells.Remove(cell.Key);
+            }
+
+
+            // Fill the gap with new cells, using the same style as the first row.
+            foreach (Cell cell in upperRow)
+            {
+                for (Int32 i = 0; i < numberOfNewRows; i++)
+                {
+                    Address newAddress = new Address(cell.CellAddress2.Column, cell.CellAddress2.Row + 1 + i); 
+                    Cell newCell = new Cell("", Cell.CellType.EMPTY, newAddress);
+                    if (cell.CellStyle != null)
+                        newCell.SetStyle(cell.CellStyle);
+                    this.Cells.Add(newAddress.GetAddress(), newCell);
+                }
+            }
+
+            // Re-add the previous cells from the copy back with a new key.
+            foreach (KeyValuePair<string,Cell> cellKeyValue in newCells)
+            {
+                this.Cells.Add(cellKeyValue.Key, cellKeyValue.Value);  //cell.Value is the cell incl. Style etc.
+            }
+        }
+
+        /// <summary>
+        /// Searches for the first occurrence of the value.
+        /// </summary>
+        /// <param name="searchValue">The value to search for.</param>
+        /// <returns>The first cell containing the searched value or Null</returns>
+        public Cell FirstCellByValue(object searchValue)
+        {
+            var cell = this.Cells.FirstOrDefault(c =>
+                c.Value?.Value != null &&
+                Equals(c.Value.Value, searchValue))
+                .Value;
+            return cell;
+        }
+
+        /// <summary>
+        /// Searches for the first occurrence of the expression.
+        /// Example: var cell = worksheet.FindCell(c => c.Value?.ToString().Contains("searchValue"));
+        /// </summary>
+        /// <param name="predicate"></param>
+        /// <returns>The first cell containing the searched value or Null</returns>
+        public Cell FirstOrDefaultCell(Func<Cell, bool> predicate)
+        {
+            return this.Cells
+                .FirstOrDefault(c => predicate(c.Value))
+                .Value;
+        }
+
+        /// <summary>
+        /// Searches for cells that contain the specified value and returns a list of these cells.
+        /// </summary>
+        /// <param name="searchValue">The value to search for.</param>
+        /// <returns>A list of cells that contain the specified value.</returns>
+        public List<Cell> CellsByValue(object searchValue)

```

Comment:
Here, it is not possible to get a list of empty cells, where the value is null (could be a useful function IMHO). This is also the cause of the previously described, possible bug.

The same applies also to the function `public Cell FirstCellByValue(object searchValue)` on line 2694